File: management-domain/Finance-department/Backend/Java/conversion-service/src/main/java/com/gogidix/finance/conversion/shared/base/BaseEntity.java
package com.gogidix.finance.conversion.shared.base;

import lombok.Data;
import org.springframework.data.annotation.CreatedBy;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;

import java.time.Instant;

/**
 * Base Entity for all MongoDB documents
 * Provides common fields for auditing
 */
@Data
public abstract class BaseEntity {

    @Id
    protected String id;

    @CreatedDate
    protected Instant createdAt;

    @LastModifiedDate
    protected Instant updatedAt;

    @org.springframework.data.annotation.LastModifiedBy
    @CreatedBy
    protected String updatedBy;
}

File: management-domain/Executive-domain-X/Executive-domain-X/Backend/Nodes/executive-realtime-service/src/shared/base/base.entity.ts
import { v4 as uuidv4 } from 'uuid';
export abstract class BaseEntity { readonly id: string; readonly createdAt: Date; readonly updatedAt: Date; constructor(props?: { id?: string; createdAt?: Date; updatedAt?: Date }) { this.id = props?.id || uuidv4(); this.createdAt = props?.createdAt || new Date(); this.updatedAt = props?.updatedAt || new Date(); } }
